File: src/main/org/nlogo/prim/etc/_isnumber.java
package org.nlogo.prim.etc ;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _isnumber
	extends Reporter
	implements org.nlogo.nvm.Pure
{
	@Override
	public Object report( final Context context ) throws LogoException
	{
		return ( args[ 0 ].report( context ) instanceof Double )
			? Boolean.TRUE
			: Boolean.FALSE ;
	}
	@Override
	public Syntax syntax()
	{
		return Syntax.reporterSyntax
			( new int[] { Syntax.TYPE_WILDCARD } ,
			  Syntax.TYPE_BOOLEAN ) ;
	}
}

File: src/main/prim/etc/_isnumber.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.core.Syntax
import org.nlogo.nvm.{ Context, Pure, Reporter }

class _isnumber extends Reporter with Pure {
  override def syntax =
    Syntax.reporterSyntax(
      right = List(Syntax.WildcardType),
      ret = Syntax.BooleanType)
  override def report(context: Context): java.lang.Boolean =
    Boolean.box(
      args(0).report(context).isInstanceOf[java.lang.Double])
}
